### ISSUE TITLE
SymfonyServiceDefinitionValidator which does not supports the new Factory syntax for definitions

### ISSUE THREAD

**Issue by craigh** [matthiasnoback/symfony-service-definition-validator#21]
see https://github.com/symfony/symfony/issues/13113

ping @stof @xabbuh

Since Dec 23, 2014 the SYmfony 2.6.1 branch is not working properly with regard to the validator service.

@stof said:

> This is a bug in Matthias\SymfonyServiceDefinitionValidator which does not supports the new Factory syntax for definitions

``` Shell
  [Matthias\SymfonyServiceDefinitionValidator\Exception\InvalidServiceDefinitionsException]                                       
  Service definition validation errors (1):                                                                                       
  - validator: Definition for class Symfony\Component\Validator\Validator has no argument for required parameter metadataFactory  

Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the post-update-cmd event terminated with an exception

  [RuntimeException]                                                         
  An error occurred when executing the "'cache:clear --no-warmup'" command.  

Exception trace:
 () at /Applications/MAMP/htdocs/core.git/src/vendor/sensio/distribution-bundle/Sensio/Bundle/DistributionBundle/Composer/ScriptHandler.php:407
 Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::executeCommand() at /Applications/MAMP/htdocs/core.git/src/vendor/sensio/distribution-bundle/Sensio/Bundle/DistributionBundle/Composer/ScriptHandler.php:115
 Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache() at phar:///usr/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:206
 Composer\EventDispatcher\EventDispatcher->executeEventPhpScript() at phar:///usr/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:174
 Composer\EventDispatcher\EventDispatcher->doDispatch() at phar:///usr/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:120
 Composer\EventDispatcher\EventDispatcher->dispatchCommandEvent() at phar:///usr/bin/composer/src/Composer/Installer.php:333
 Composer\Installer->run() at phar:///usr/bin/composer/src/Composer/Command/UpdateCommand.php:134
 Composer\Command\UpdateCommand->execute() at phar:///usr/bin/composer/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:252
 Symfony\Component\Console\Command\Command->run() at phar:///usr/bin/composer/vendor/symfony/console/Symfony/Component/Console/Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/bin/composer/vendor/symfony/console/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at phar:///usr/bin/composer/src/Composer/Console/Application.php:146
 Composer\Console\Application->doRun() at phar:///usr/bin/composer/vendor/symfony/console/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at phar:///usr/bin/composer/src/Composer/Console/Application.php:83
 Composer\Console\Application->run() at phar:///usr/bin/composer/bin/composer:43
 require() at /usr/bin/composer:25
```

**Comment by craigh** [comment]
maybe having the same trouble again here :sob: 

It was working until today I did a composer update:
```Shell
craig% composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)                 
  - Installing matthiasnoback/symfony-console-form (dev-master 3ce9795)
    Cloning 3ce97954c3f4e65291fbb74fb2f41d7a0b32d921

  - Updating monolog/monolog dev-master (21259d5 => b18a6ef)
    Checking out b18a6eff04b15184375a6875ef2aa54bc5a7c9f2

  - Updating vakata/jstree dev-master (2f630b4 => fe24228)
    Checking out fe24228e1b2842501f595fa2838d8da44eaff0bd

  - Updating symfony/symfony 2.6.x-dev (77b68b3 => d00a435)
    Checking out d00a43554c3997d8a9c861f6efd16b7268be2f62

Writing lock file
Generating autoload files
Compiling component files
Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the post-update-cmd event terminated with an exception

  [RuntimeException]                                                         
  An error occurred when executing the "'cache:clear --no-warmup'" command.                                                                          
```

so I did:
```Shell
craig% composer update -vvv
...
Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the post-update-cmd event terminated with an exception

  [RuntimeException]                                                         
  An error occurred when executing the "'cache:clear --no-warmup'" command.  

Exception trace:
 () at /Applications/MAMP/htdocs/core.git/src/vendor/sensio/distribution-bundle/Sensio/Bundle/DistributionBundle/Composer/ScriptHandler.php:407
 Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::executeCommand() at /Applications/MAMP/htdocs/core.git/src/vendor/sensio/distribution-bundle/Sensio/Bundle/DistributionBundle/Composer/ScriptHandler.php:115
 Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache() at phar:///usr/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:206
 Composer\EventDispatcher\EventDispatcher->executeEventPhpScript() at phar:///usr/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:174
 Composer\EventDispatcher\EventDispatcher->doDispatch() at phar:///usr/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:120
 Composer\EventDispatcher\EventDispatcher->dispatchCommandEvent() at phar:///usr/bin/composer/src/Composer/Installer.php:336
 Composer\Installer->run() at phar:///usr/bin/composer/src/Composer/Command/UpdateCommand.php:136
 Composer\Command\UpdateCommand->execute() at phar:///usr/bin/composer/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:252
 Symfony\Component\Console\Command\Command->run() at phar:///usr/bin/composer/vendor/symfony/console/Symfony/Component/Console/Application.php:874
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/bin/composer/vendor/symfony/console/Symfony/Component/Console/Application.php:195
 Symfony\Component\Console\Application->doRun() at phar:///usr/bin/composer/src/Composer/Console/Application.php:146
 Composer\Console\Application->doRun() at phar:///usr/bin/composer/vendor/symfony/console/Symfony/Component/Console/Application.php:126
 Symfony\Component\Console\Application->run() at phar:///usr/bin/composer/src/Composer/Console/Application.php:83
 Composer\Console\Application->run() at phar:///usr/bin/composer/bin/composer:43
 require() at /usr/bin/composer:25
```

please help :cry: @xabbuh @stof @matthiasnoback

**Comment by craigh** [comment]
I tried to force various releases of symfony (2.6.1, 2.6.2, 2.6.3) but that didn't seem to make a difference :disappointed:

**Comment by matthiasnoback** [comment]
@craigh I don't see anything related to this library in the error message. Did you update this package already to its latest version? If not, you should try to manually clear the cache by removing all the files and then manually run `app/console cache:clear`. If the error still occurs, please report here again. 

Also, would you open a separate ticket for this please?

**Comment by craigh** [comment]
@matthiasnoback :smile: 

I'm so very sorry to have wasted your time :hourglass: 

of course you are right, the library is not in the error message. I was assuming the problem was similar because I was getting another error and well... I'm not that good at this stuff yet.

Turns out, I was a little too excited to try your new `symfony-console-form` component and had an incomplete Command class that was present before the library was ready to deal with it and that was causing the error :8ball: 

Once I did as you suggested and also removed my wayward class, everything returned to normal.

thank you for your time!

(nothing to see here... move along)
